### PR TITLE
Make github hide generated files in diffs

### PR DESCRIPTION
**Pull request by mikekap**
**What this PR does / why we need it**:
https://github.com/github/linguist#generated-code says to add `linguist-generated=true` to any files that you don't want to see diffs in. IMO this will make PRs a little easier to review.

e.g. the top half of https://github.com/kubernetes/kubernetes/pull/53988 should disappear

**Which issue this PR fixes** 

**Special notes for your reviewer**:

**Release note**: NONE

### CODE REVIEW

**Comment by cblecker** [comment]
/retest
/release-note-none

I like this change.
@kubernetes/api-reviewers any issues or objections to this?

**Comment by liggitt** [comment]
can the diffs be expanded if needed? in PRs that actually change generation, it's necessary to be able to review the output

File: .gitattributes [hunk 1]
```diff
@@ -1,9 +1,10 @@
...
-**/generated.pb.go -diff
-**/generated.proto -diff
-**/types_swagger_doc_generated.go -diff
-docs/api-reference/** -diff
+**/zz_generated.*.go -diff linguist-generated=true
+**/types.generated.go -diff linguist-generated=true
+**/generated.pb.go -diff linguist-generated=true
+**/generated.proto -diff linguist-generated=true
+**/types_swagger_doc_generated.go -diff linguist-generated=true
+docs/api-reference/** -diff linguist-generated=true
```
Comment (cblecker) [inline .gitattributes]: I think this should be marked as documentation, rather than generated.
https://github.com/github/linguist#documentation

Comment (liggitt) [reply .gitattributes]: or both generated and documentation

**Comment by cblecker** [comment]
@liggitt I thought so, but [this sentence](https://github.com/github/linguist#generated-code) gave me pause: `As an added bonus, unlike vendored and documentation files, these files are suppressed in diffs.`

File: .gitattributes [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
-**/zz_generated.*.go -diff
-**/types.generated.go -diff
-**/generated.pb.go -diff
-**/generated.proto -diff
-**/types_swagger_doc_generated.go -diff
-docs/api-reference/** -diff
+**/zz_generated.*.go -diff linguist-generated=true
+**/types.generated.go -diff linguist-generated=true
+**/generated.pb.go -diff linguist-generated=true
+**/generated.proto -diff linguist-generated=true
```
Comment (lavalamp) [inline .gitattributes]: We may actually want to see proto diffs.

**Comment by mikekap** [comment]
From my reading of `linguist-documentation` it looks like linguist already recognizes files in `docs/` to be documentation based on the path; see: https://github.com/github/linguist/blob/master/lib/linguist/documentation.yml . I can add it for completeness if that helps.

File: .gitattributes (hunk 1 above)
Comment (lavalamp) [reply .gitattributes]: I think we want to see the documentation changes in review.

File: .gitattributes [hunk 3]
```diff
@@ -1,9 +1,10 @@
...
+**/types_swagger_doc_generated.go -diff linguist-generated=true
+docs/api-reference/** -diff linguist-generated=true
+api/swagger-spec/*.json -diff linguist-generated=true
```
Comment (lavalamp) [inline .gitattributes]: I'm not 100% sure but I think we want to see this in review.

**Review by lavalamp — COMMENTED** [review]
I marked the ones I'm not sure about; the others I'm happy to lgtm. I recommend making a second PR so we can get this in for the non-controversial ones immediately.

**Comment by mikekap** [comment]
@liggitt the diffs can definitely be expanded. It looks like this:

screen shot 2017-10-30 at 2 26 36 pm

File: .gitattributes (hunk 3 above)
Comment (mikekap) [reply .gitattributes]: Are you...sure? This opens up several hundred lines of json on the top of https://github.com/kubernetes/kubernetes/pull/53988/files at least. Are most reviewers actually reading those json dumps? I was under the impression most folks look at the `.go` files where these are defined in more readable form. 

In either case - these aren't lost from the PR - they just default to collapsed.

Comment (lavalamp) [reply .gitattributes]: ah OK I am ok with hiding that by default.

**Comment by mikekap** [comment]
So what’s the path for getting this in? @lavalamp I sort of took our last discussion to mean there’s no need to hide anything given you can expand manually at CR time. Does that sound right to you?

**Comment by liggitt** [comment]
if it just collapses diffs by default in PR review, this LGTM as is

**Comment by lavalamp** [comment]
I took a look again, if you add back in the generated proto files I'll LGTM. Those are short.

**Comment by mikekap** [comment]
@lavalamp done

**Pull request merged**